### ISSUE TITLE
1.6.2-beta.10 - Error in babel minify - Cannot read property 'isPure' of null while minifying

### ISSUE THREAD

**Issue by filipenevola** [meteor/meteor#9711]
Hello, just to keep track of this bug in babel minify https://github.com/babel/minify/issues/803

`Cannot read property 'isPure' of null while minifying`

Because of it, I'm not able to run my app in production mode (galaxy deploy or bundle-visualizer build)

I think it's only fixed on master branch, there is a way to Meteor point to it in beta.11 or we need to wait for the next release of babel minify?

my error stack trace: https://gist.github.com/filipenevola/716441bd69fd9f8c5ad77501f0a1b9f3

**Comment by hwillson** [comment]
Thanks for reporting this @filipenevola. We'll likely wait until this is bundled in the next Babel minify release. That being said, Babel minify is only used as a fallback if minification fails using `uglify-es`. It would be great to know why `uglify-es` minification is failing for you. Any chance you could provide a reproduction?

**Comment by filipenevola** [comment]
@hwillson how can I see these details? I only see logs related to babel-minify, even using METEOR_PROFILE flag.

I need to understand what is failing in `uglify-es` to create a reproduction.

**Comment by filipenevola** [comment]
@hwillson I saw that you labeled it as "has-workaround", what is the workaround?

**Comment by hwillson** [comment]
> how can I see these details?

Any chance you could setup a [developer checkout](https://github.com/meteor/meteor/blob/devel/DEVELOPMENT.md)? If so, you could then just `console.log` the `uglify-es` error in [minifier-js/minifier.js](https://github.com/meteor/meteor/blob/a770e9959fec4e499a8f8c01dac5c211808c72e3/packages/minifier-js/minifier.js). 

> I need to understand what is failing in uglify-es to create a reproduction.

That would definitely help, but you could also start with your existing app, and start removing code/packages until it starts working again. That could help you pinpoint where the issue is, and help you create a new focused repro that shows the problem. Painful I know, but this approach often works.

> I saw that you labeled it as "has-workaround", what is the workaround?

This is a tricky one. I marked it as `Severity:has-workaround` because this issue isn't considered a production blocker (yet), since it's being reported against a beta release. Technically there are workarounds (don't use the beta release, remove `standard-minifier-js` from your app, find the offending code and adjust/remove it, etc.), but they're not great workarounds, especially if this is for a production app. To be honest I've never really liked the `Severity:has-workaround` label, as it doesn't really add much to the issue triage process. If an issue has a workaround great - let's document it in the issue, then decide if the issue should be kept open or closed. Adding a `has-workaround` label doesn't achieve much past that decision.

Long story short, I'll re-classify. Thanks!

**Comment by juvyc** [comment]
No update yet on this?

**Comment by Lucbug** [comment]
Interested in an update too.

**Comment by dealogikal** [comment]
Update on this please? thanks a lot!

**Comment by hwillson** [comment]
Hi all - until this gets fixed upstream, here's a quick workaround:

1. Grab a copy of the meteor source: `git clone https://github.com/meteor/meteor.git`
2. Recursively copy Meteor's `packages/minifier-js` folder into a `packages` directory in your app.
3. In your app, update your `packages/minifier-js/minifier.js` file to disable `babel/minify`'s `transform-remove-undefined` plugin:

```js
  ...
  } catch (e) {
    // Although Babel.minify can handle a wider variety of ECMAScript
    // 2015+ syntax, it is substantially slower than UglifyJS, so we use
    // it only as a fallback.
    var options = Babel.getMinifierOptions({
      inlineNodeEnv: NODE_ENV,
      
      //*** ADD THIS      
      // Disable the `transform-remove-undefined` plugin
      removeUndefined: false,      
      //***
      
    });
    result.code = Babel.minify(source, options).code;
    result.minifier = 'babel-minify';
  }
  ...
```

**Comment by nikkiczx** [comment]
@hwillson The workaround doesn't seem to work for me. I've tried creating a `packages` directory in my app since there isn't one. I've tried copying it into the existing `packages` directory in the global Meteor directory. Neither works. 

What am I doing wrong? I'm on Meteor 1.7.0.1 and I can't deploy to Galaxy. Is minifying crucial or can I remove standard-minifier-js until its resolved and still be able to deploy to Galaxy?

**Comment by adamgins** [comment]
HI all, I have tried with 1.7.1beta31 an still can't use the standard minifiers.  any update on this pls?

**Comment by filipenevola** [comment]
Closing this as the error is not happening anymore on Meteor recent versions.
